api/client/identity.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"os"
+
+	"github.com/gravitational/trace"
+)
+
+// IdentityFile represents the basic components of an identity file.
+type IdentityFile struct {
+	PrivateKey []byte
+	Certs      struct {
+		SSH []byte
+		TLS []byte
+	}
+	CACerts struct {
+		SSH [][]byte
+		TLS [][]byte
+	}
+}
+
+// TLS returns the identity file's associated tls config.
+func (idf *IdentityFile) TLS() (*tls.Config, error) {
+	cert, err := tls.X509KeyPair(idf.Certs.TLS, idf.PrivateKey)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	pool := x509.NewCertPool()
+	for i, caCerts := range idf.CACerts.TLS {
+		if !pool.AppendCertsFromPEM(caCerts) {
+			return nil, trace.BadParameter("identity file contains invalid TLS CA cert (#%v)", i+1)
```

Comment:
Is this user-facing? Mention that it's the index in brackets (if this is a 1-to-1 mapping from a file) or a prefix so she can identify the key in question?